agent/engine/docker_task_engine.go

```diff
@@ -141,7 +172,7 @@ func (engine *DockerTaskEngine) SetDockerClient(client DockerClient) {
 
 // MustInit blocks and retries until an engine can be initialized.
 func (engine *DockerTaskEngine) MustInit() {
-	if engine.client != nil {
+	if (engine.client != nil) && (engine.s3Client != nil) {
```

Comment:
The methods that initialize each client both check if they are already initialized.